Geometry/ForwardGeometry/src/TotemGeometry.cc

```diff
@@ -0,0 +1,65 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Author:
+*   Laurent Forthomme
+*
+****************************************************************************/
+
+#include "Geometry/ForwardGeometry/interface/TotemGeometry.h"
+
+TotemGeometry::TotemGeometry(const DetGeomDesc* dgd) {
+  std::deque<const DetGeomDesc*> buffer;
+  buffer.emplace_back(dgd);
+  while (!buffer.empty()) {
+    // get the next item
+    const auto* desc = buffer.front();
+    buffer.pop_front();
+    browse(desc, false);
+  }
+}
+
+void TotemGeometry::browse(const DetGeomDesc*& parent, bool in_t2) {
+  if (parent->name() == "TotemT2")
+    in_t2 = true;  // define the mother volume for all children
+  if (in_t2)
+    browseT2(parent);
+  // start the recursive browsing
+  for (const auto& child : parent->components())
+    browse(const_cast<const DetGeomDesc*&>(child), in_t2);
```

Comment:
Not necessarily. In a follow-up push I adapted the `TotemGeometry::browse` method to accept a const pointer instead of a const reference to a pointer, and thus the recursive browsing is simplified.